notebooks/integrations/cohere/cohere-elasticsearch.ipynb

```diff
@@ -0,0 +1,562 @@
+    "\n",
+    "This example uses the [SciFact](https://huggingface.co/datasets/mteb/scifact) data\n",
+    "set that you can find on HuggingFace."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c71b8367",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = \"https://huggingface.co/datasets/mteb/scifact/raw/main/corpus.jsonl\"\n",
+    "\n",
+    "# Fetch the JSONL data from the URL\n",
+    "response = requests.get(url)\n",
+    "response.raise_for_status()  # Ensure we notice bad responses\n",
+    "\n",
+    "# Split the content by new lines and parse each line as JSON\n",
+    "data = [json.loads(line) for line in response.text.strip().split(\"\\n\") if line]\n",
+    "\n",
+    "# Change `_id` key to `id` as `_id` is a reserved key in Elasticsearch.\n",
+    "for item in data:\n",
+    "    if \"_id\" in item:\n",
+    "        item[\"id\"] = item.pop(\"_id\")\n",
+    "\n",
+    "# Prepare the documents to be indexed\n",
+    "documents = []\n",
+    "for line in data:\n",
+    "    data_dict = line\n",
+    "    documents.append(\n",
+    "        {\n",
+    "            \"_index\": \"cohere-embeddings\",\n",
+    "            \"_source\": data_dict,\n",
+    "        }\n",
+    "    )\n",
+    "\n",
+    "# Use the bulk endpoint to index\n",
+    "helpers.bulk(client, documents)\n",
+    "\n",
+    "print(\"Data ingestion completed, text embeddings generated!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9332078a",
+   "metadata": {},
+   "source": [
+    "Your index is populated with the SciFact data and text embeddings for the text\n",
+    "field."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "610035af",
+   "metadata": {},
+   "source": [
+    "## Hybrid search\n",
+    "\n",
+    "Let's start querying the index!\n",
+    "\n",
+    "The code below performs a hybrid search. The `kNN` query computes the relevance\n",
+    "of search results based on vector similarity using the `text_embedding` field.\n",
+    "The lexical search query uses BM25 retrieval to compute keyword similarity on\n",
+    "the `title` and `text` fields."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ab735ef0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query = \"What is biosimilarity?\"\n",
+    "\n",
+    "response = client.search(\n",
+    "    index=\"cohere-embeddings\",\n",
+    "    size=100,\n",
+    "    knn={\n",
+    "        \"field\": \"text_embedding\",\n",
+    "        \"query_vector_builder\": {\n",
+    "            \"text_embedding\": {\n",
+    "                \"model_id\": \"cohere_embeddings\",\n",
+    "                \"model_text\": query,\n",
+    "            }\n",
+    "        },\n",
+    "        \"k\": 10,\n",
+    "        \"num_candidates\": 50,\n",
+    "    },\n",
+    "    query={\"multi_match\": {\"query\": query, \"fields\": [\"text\", \"title\"]}},\n",
+    ")\n",
```

Comment:
This could be something I don't know, but I thought when running two queries as in this example you have to add RRF to combine the results into a single list. Is that optional? And if it is, how are the results from the two queries combined?

Reply:
`RRF` is optional and I _think_ the results are combined based on the `score` you get for the hits in case you didn't specify it. If we want to add `rrf` to the query, the `window_size` parameter needs to be tweaked to be greater than or equal to 100 (the `size` parameter) as its default value is `10`. I felt it would require some explanation that might make the notebook harder to comprehend. For this reason, I  decided not to add a specific ranking method.

Reply:
This is fine, I just did not know there was a default algorithm to combine the results.